web/src/engine/websites/CopyManga.ts

```diff
@@ -104,47 +87,50 @@ export default class extends DecoratableMangaScraper {
     }
 
     public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
-        const chapterList = [];
-        for (let page = 0, run = true; run; page++) {
-            const chapters = await this.GetChaptersFromPage(manga, page);
-            chapters.length > 0 ? chapterList.push(...chapters) : run = false;
-        }
-        return chapterList;
-    }
 
-    private async GetChaptersFromPage(manga: Manga, page: number): Promise<Chapter[]> {
-        try {
-            const request = this.CreateApiRequest(`comic/${manga.Identifier}/group/default/chapters?limit=500&offset=${page * 500}`);
-            const data = await FetchJSON<APIResponse<APIResultList<APIChapter>>>(request);
-            return data.results.list.map(item => new Chapter(this, manga, item.uuid, item.name.trim()));
-        } catch (error) {
-            return [];
-        }
+        const request = this.CreateApiRequest(`/comicdetail/${manga.Identifier}/chapters`);
+        const { results } = await FetchJSON<APIResponse<string>>(request);
+        const { groups: { default: { chapters } } } = await this.Decrypt<APIChapters>(results);
+        return chapters.map(chapter => new Chapter(this, manga, chapter.id, chapter.name.trim()));
     }
 
     public override async FetchPages(chapter: Chapter): Promise<Page[]> {
-        const request = this.CreateApiRequest(`comic/${chapter.Parent.Identifier}/chapter2/${chapter.Identifier}?platform=3`);
-        const data = await FetchJSON<APIResponse<APIPages>>(request);
-        const imageUrls = data.results.chapter.contents.map(item => item.url);
-        const imageOrder = data.results.chapter.words;
-        const orderedPages : string[] = [];
-        imageOrder.forEach((order, index) => {
-            orderedPages[order] = imageUrls[index];
-        });
-        return orderedPages.map(page => new Page(this, chapter, new URL(page)));
+
+        const [dataElement] = await FetchCSS<HTMLDivElement>(new Request(new URL(`/comic/${chapter.Parent.Identifier}/chapter/${chapter.Identifier}`, this.URI)), '.imageData');
+        const imageData = dataElement.getAttribute('contentKey');
+        const images = await this.Decrypt<APIPage[]>(imageData);
+        return images.map(image => new Page(this, chapter, new URL(image.url)));
+
+    }
+
+    private async Decrypt<T>(encryptedData: string, keyString: string = this.defaultKey): Promise<T> {
+
+        const key = Buffer.from(keyString, 'utf-8');
+        const iv = Buffer.from(encryptedData.substring(0x0, 0x10), 'utf-8');
```

Comment:
Personally in this case i prefer to use `0` and `16` instead of their hex representation.
Same for `cipher`.

Reply:
Ok, I dont have any preference, its just copypasted anyway .

Reply:
Avoid placing 1:1 copy/pasted functions from the website into the class as it may interfere with the `this` scope.
It would be preferred to place them outside of the class.